### DISCUSSION TITLE
Questions regarding 2 homework

### DISCUSSION THREAD

**Discussion by OlgaKuzminichna** [cafawo/FinancialDataAnalytics#10] [Q&A] [answered]
# My questions regarding second homework: 

1. Is it fine to publish homework in py format rather than Ipub?
2. Task 2 with histogram, what could be the best advice on picking right bins
3. Why we don't smooth the plot with returns, 
  for me it seems to be irregular fluctuations and I would prefer to use rolling for a better picture?
4. Task with heat map. I got confused why it is:
```
 returns.iloc[:10, :10].corr() 
```
but not 
```
returns.iloc[:, :10].corr()
```
The first option means we take first 10 companies and only first 10 dates, so we ignore another 240 dates
<img src="https://user-images.githubusercontent.com/120052171/236870346-62bb44c7-7ba9-4b61-ba00-50d46da76e40.png " width="300" height="300">

5. Why SIEGn.DE  .GDAXI is not picked as highest stock pair as 83 is not 85
![image](https://user-images.githubusercontent.com/120052171/236869789-1ab40ce6-d893-4f0e-b21e-3b24f2f88fc1.png)

**Comment by OlgaKuzminichna** [comment, marked as answer]
The answers for my questions:

1. Fine
2. Mostly depends on research question, but there is a general advice available on the internet
3. It is fine to smooth for a homework, it could result in the loss of variability and heteroscedasticity
4. it is typo- correct option is  returns.iloc[:, :10].corr()
alternatively we could calculate correlation first and then filter rows and columns by 10
5. Because  .GDAXI is not a stock

**Comment by cafawo** [comment]
Regarding Q2, the optimal number of bins in a Histogram:

You could consider the [Freedman–Diaconis rule](https://en.wikipedia.org/wiki/Freedman%E2%80%93Diaconis_rule), which is based on this paper: https://link.springer.com/article/10.1007/BF01025868

Technically Freedman–Diaconis gives you the optimal bin with $h$, if you want the number of bins $n$ you can covert by

$$n = \frac{\max - \min}{h}$$
